### ISSUE TITLE
Bug: [space-infix-ops] false positive since version 5.27.0

### ISSUE THREAD

**Issue by paztis** [typescript-eslint/typescript-eslint#5133]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=FAOwhgtgpgzgDmAxlABDANgS2QESgMxQG9gUUoAPOAewCcAXFTEeqW-JVASQGV6xWxFAF9go4ABMoidGFqpE1EDEaJ5AqDyjpp9OgC4UYEAE8A3MGCUaDFFI4BXdKtkwYKLTsR7a7kmQxsTW1dOgAKRAdaeRY+DUNA3AIAOl5+VgBKBKwk-FS4wSIUeXookBRI6KhY9KgzEUsyKH4AcxhPUNpDMJV4tByoPDy0jQyUAF4APhQAJWk6CQAeFVpmFoAaNHpVkBbp8dIyCvVWDu9w+gALTBhkxOCvH02egbGp-qDk5rA2jItRIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKWQBwEMBjRAWgEsA7AM3IA9SB7PVDSR66yAGnG0gACAFwCeeFEWjk8Q0inhUhAenzEyVWg2aswAbT7YoiaNEbQeB7FkP9FAZgBMACUXow1AvGSJLEAL6WALp8AX5AA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
namespace sliceDef {
  export interface IState { }
}

declare const createSelector: any;

export default class Selectors {
  sliceSelector(currentState: sliceDef.IState): sliceDef.IState { return currentState; }

  etagsSelector: (state: sliceDef.IState) => Record<string, string> =
    createSelector(this.sliceSelector, (slice) => slice.etags);
}
```

### ESLint Config

```javascript
{
  "rules": {
    "space-infix-ops": "off",
    "@typescript-eslint/space-infix-ops": [
      "error",
      {
        "int32Hint": false
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

no error

### Actual Result

  11:26  error  Operator ':' must be spaced               @typescript-eslint/space-infix-ops

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.27.0` |
| `@typescript-eslint/parser`            | `5.27.0` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.16.0` |
| `node`                                 | `16.5.0` |

**Comment by CyanSalt** [comment]
It looks like the latest alpha release now fixes the problem in most cases, but does not include those cases that contain `extends`, e.g.

```typescript
type Foo<T> = T extends { bar: string } ? string : number
                             ^
```

```typescript
type Foo<T> = T extends (bar: string) => void ? string : number
                            ^
```

**Comment by paztis** [comment]
Canary is released on npm ?

**Comment by bradzacher** [comment]
Please use the issue search before filing an issue. Closed issues are a must when searching.

**Comment by paztis** [comment]
Wait. I just discover this defect yesterday and was thinking it was due to this 2 days ago commit. 
Don't say I don't search on issues before posting.

Now even if the defect is closed it is still not released... When is it supposed to be released ?

**Comment by armano2** [comment]
you can always use canary versions, eg, 5.27.1-alpha.5 witch contains fix for this issue

as for normal release, we have them at beginning of the week (~monday)
